### ISSUE TITLE
How did you make the Ballons Translator project? 

### ISSUE THREAD

**Issue by selori** [dmMaze/BallonsTranslator#516]
First of all, I'm an ignorant person for asking this question, I know that.Anyway, there are a few things I want to learn.
As you said, the project depends on manga-image-translator as the backend.Of course, I am not saying this to disrespect your efforts, I really respect you very much. 
How did you make this project using this repo?For example, setting text blocks on the selected image. Here's how you do this and then coloring the text and finally saving and exporting. Maybe I'm exaggerating, but can you explain everything briefly?

( Actually, I'm aiming to make something like a web adaptation of the Ballons Translator application. There is already a project called cortans that is dependent on manga-image-translator. However, I want to try what I can do myself. I have a social and modern anime site and I am developing it. However, I also need a manga version and a feature so that I can open it abroad. I don't want to advertise, so I won't specify the site link unless you want, and at the moment it's only in Turkish. )

Thank you in advance for your feedback :)

**Comment by bropines** [comment]
> First of all, I'm an ignorant person for asking this question, I know that.Anyway, there are a few things I want to learn. As you said, the project depends on manga-image-translator as the backend.Of course, I am not saying this to disrespect your efforts, I really respect you very much. How did you make this project using this repo?For example, setting text blocks on the selected image. Here's how you do this and then coloring the text and finally saving and exporting. Maybe I'm exaggerating, but can you explain everything briefly?
> 
> ( Actually, I'm aiming to make something like a web adaptation of the Ballons Translator application. There is already a project called cortans that is dependent on manga-image-translator. However, I want to try what I can do myself. I have a social and modern anime site and I am developing it. However, I also need a manga version and a feature so that I can open it abroad. I don't want to advertise, so I won't specify the site link unless you want, and at the moment it's only in Turkish. )
> 
> Thank you in advance for your feedback :)

Well, I’m now creating the same service based on MIT. In fact, you could simply write UA in any language and communicate via API with MIT. Or you can directly use MIT methods as a Python library. The main thing here is to be able to work with packages. If you want simplicity, write an API client that will also translate through MIT and edit through your program. If you want something more complicated, take, as I already said, methods from the MIT library

PS. MIT - Manga Image Translate

**Comment by selori** [comment]
> Well, I’m now creating the same service based on MIT. In fact, you could simply write UI in any language and communicate via API with MIT. Or you can directly use MIT methods as a Python library. The main thing here is to be able to work with packages. If you want simplicity, write an API client that will also translate through MIT and edit through your program. If you want something more complicated, take, as I already said, methods from the MIT library
> 
> PS. MIT - Manga Image Translate

Firstly thx for reply.
I will be open , I'm not very focused on Python nowadays, but I want to refocus on Python right now. My knowledge of Python is weak (not very weak, but I had previously only concentrated on Django in Python and other some libraries pygame, flask).
If you want, I can help you with your project, but I can help you with the UI aspect and I think it will be html css js based. I can also connect the API. Anyway, can you at least explain to me the methods in MIT? I haven't been able to run the code on my computer yet. I think there is a problem with the libraries downloaded from pip, but hopefully I will solve it. I couldn't quite figure out where the methods are in the MIT library. I looked at the UIs with QT, but I still couldn't figure it out, did I look in the wrong place?

**Comment by bropines** [comment]
> Firstly thx for reply.
> I will be open , I'm not very focused on Python nowadays, but I want to refocus on Python right now. My knowledge of Python is weak (not very weak, but I had previously only concentrated on Django in Python and other some libraries pygame, flask).
> If you want, I can help you with your project, but I can help you with the UI aspect and I think it will be html css js based. I can also connect the API. Anyway, can you at least explain to me the methods in MIT? I haven't been able to run the code on my computer yet. I think there is a problem with the libraries downloaded from pip, but hopefully I will solve it. I couldn't quite figure out where the methods are in the MIT library. I looked at the UIs with QT, but I still couldn't figure it out, did I look in the wrong place?

Well look. In fact, the entire MIT is not even a library, but a project with 5 modules. That is
1.TextDetector
2.Ocr
3.Translate
4.Inpaint
5. Overwrite translated text

Of which 3 work within our program: TD, OCR, Inpaint. They work through pre-trained models. Scripts (modules) that use them are registered here (in the appropriate folders)
![image](https://github.com/user-attachments/assets/96817eb2-1e0a-464d-b96f-c49fb104358b)


In fact, you can also import these modules and request functions in them. But judging by your experience, you first need to learn how to write both programs with libraries, and also write a library (module) yourself, which you will embed in an external script. In fact, Python imports the same Python files but with ready-made functions, and then simply uses them.

**Comment by selori** [comment]
> Bakın. Aslında, tüm MIT bir kütüphane bile değil, 5 modülden oluşan bir projedir. Bunlar 1.TextDetector 2.Ocr 3.Translate 4.Inpaint 5. Overwrite translate text
> 
> Bunlardan 3'ü programımız içinde çalışır: TD, OCR, Inpaint. Önceden eğitilmiş modeller üzerinden çalışırlar. Bunları kullanan betikler (modüller) burada (uygun klasörlerde) kayıtlıdır ![görüntü](https://private-user-images.githubusercontent.com/57861007/348498534-96817eb2-1e0a-464d-b96f-c49fb104358b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA4OTk0MzcsIm5iZiI6MTcyMDg5OTEzNywicGF0aCI6Ii81Nzg2MTAwNy8zNDg0OTg1MzQtOTY4MTdlYjItMWUwYS00NjRkLWI5NmYtYzQ5ZmIxMDQzNThiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzEzVDE5MzIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YWQ2MmY2ZmMzNWI1MTdjYjdmNDM4ZmM2ZTgzNjk0MDNhODJkN2I3NDMzZDFlMzM0MWM4ZDZiZjkxNWNlMGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WXGwFfoLhe2yIa0e3pKW8UEeHbHF5NRctyQfc9HD1Yw)
> 
> Aslında, bu modülleri içe aktarabilir ve içlerinde fonksiyonlar isteyebilirsiniz. Ancak deneyiminize bakılırsa, öncelikle her iki programı da kütüphanelerle nasıl yazacağınızı ve ayrıca harici bir betiğe gömeceğiniz bir kütüphaneyi (modülü) kendiniz nasıl yazacağınızı öğrenmeniz gerekir. Aslında, Python aynı Python dosyalarını hazır fonksiyonlarla içe aktarır ve sonra bunları basitçe kullanır.

Well now I realize I didn't actually check the code. I looked at how you did it through the UI and how the models worked, but I missed the actual files. I was confused because the UI was qt based and I wasn't very familiar with it. Now I looked at MIT's code and I think I understand. For now, I will start by evaluating the code, creating the project's algorithm in my mind, and making the UI.I will make the UI html css based because I am used to it, so instead of using vue or react I will make it directly html css based. How do you create the web project you mentioned? For example, what will you base the UI on?

**Comment by bropines** [comment]
we can ask dmmaze to move this to discussion and continue communication there. and we also have a discord, I don't speak English well, but I'll try to help you there.

**Comment by selori** [comment]
Yes, we can communicate better if we go to discord. Not knowing English is the same for me :) 
So what do I do now? When you say Discord server, do you mean manga image translator's server?

**Comment by bropines** [comment]
> Yes, we can communicate better if we go to discord or moving to discussion. Not knowing English is the same for me :)
> So what do I do now? When you say Discord server, do you mean manga image translator's server?


https://github.com/zyddnys/manga-image-translator
>Translate texts in manga/images.
[中文说明](https://github.com/zyddnys/manga-image-translator/blob/main/README_CN.md) | [Change Log](https://github.com/zyddnys/manga-image-translator/blob/main/CHANGELOG.md)
Join us on discord https://discord.gg/Ak8APNy4vb

**Comment by selori** [comment]
Okay, I already joined the server a few days ago. But let's talk about what is your username? Anyway, I'll give you mine, you can ping me: seloreis2101
